### PR TITLE
[MAINT] Add 'fits per second' control to hpi.

### PR DESCRIPTION
**Pull request by gabrielbmotta**
Preamble:
This is not a permanent fix by any means, this is a quick fix to have the continuous hpi in its current state more easily debugable during runtime (with different setups). This is a control for making iNumberOfFitsPerSecond variable during runtime, which was found to be a temporary solution to getting the continuous hpi fitting working on different incoming frequencies and data chunk sizes. iNumberOfFitsPerSecond does not set the number of fits per second, it is used to set the sample window used to perform the continuous fit, but the fit still seems to occur at whatever rate the data comes in. Treat this as a tuning variable, find a setting that works and leave it there. With that being said:

- Add controls for setting fits per seocnd in HpiSettingsView
- Make iNumberOfFitsPerSecond changeable during runtime

### CODE REVIEW

**Review by juangpc — DISMISSED** [review]

**Comment by juangpc** [comment]
It is great that it is saved in settings. 
One doubt. Since the variable ```m_iNumberOfFitsPerSecond``` doesn't really affect the actual number of fits per second done, shouldn't we change at least the control name in the gui?

**Comment by gabrielbmotta** [comment]
> It is great that it is saved in settings.
> One doubt. Since the variable `m_iNumberOfFitsPerSecond` doesn't really affect the actual number of fits per second done, shouldn't we change at least the control name in the gui?

Sure, what do you propose?

**Comment by juangpc** [comment]
Not sure, something like: fittingWindowSize or numSamplesToFit. to be set explicitly in samples by the (smooth) operator...

**Comment by juangpc** [comment]
> Not sure, something like: fittingWindowSize or numSamplesToFit. to be set explicitly in samples by the (smooth) operator...

Let me push this. I'll do it.

**Review by juangpc — APPROVED** [review]

File: libraries/disp/viewers/hpisettingsview.cpp [hunk 1]
```diff
@@ -242,6 +253,7 @@ void HpiSettingsView::loadSettings()
     m_pUi->m_checkBox_useComp->setChecked(settings.value(m_sSettingsPath + QString("/HpiSettingsView/useCOMP"), false).toBool());
     m_pUi->m_checkBox_continousHPI->setChecked(settings.value(m_sSettingsPath + QString("/HpiSettingsView/continousHPI"), false).toBool());
     m_pUi->m_doubleSpinBox_maxHPIContinousDist->setValue(settings.value(m_sSettingsPath + QString("/HpiSettingsView/maxError"), 10.0).toDouble());
+    m_pUi->m_spinBox_samplesToFit->setValue(settings.value(m_sSettingsPath + QString("/HpiSettingsView/fittingWindowSize"), 1000).toInt());
```
Comment (gabrielbmotta) [inline libraries/disp/viewers/hpisettingsview.cpp:256]: It looks like here the default is set to 1000, whereas in the constexpr at the top of the file you set the default to 100. Is this intentional?

Comment (juangpc) [reply libraries/disp/viewers/hpisettingsview.cpp:256]: Nope. I'll push a fix. It is a by product of me + the fact that some settings for a model are updated/initialized in the view!?

**Review by juangpc — DISMISSED** [review]

**Review by juangpc — APPROVED** [review]

**Pull request merged**